DSN/src/main/java/org/gfa/dsn/controller/UserController.java

```diff
@@ -0,0 +1,68 @@
+package org.gfa.dsn.controller;
+
+import org.gfa.dsn.dto.*;
+import org.gfa.dsn.repository.UserRepository;
+import org.gfa.dsn.service.JwtService;
+import org.gfa.dsn.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1")
+public class UserController {
+    private final UserService userService;
+    private final UserRepository userRepository;
+    private final AuthenticationManager authenticationManager;
+    private final JwtService jwtService;
+
+    @Autowired
+    public UserController(UserService userService, UserRepository userRepository, AuthenticationManager authenticationManager, JwtService jwtService) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+        this.authenticationManager = authenticationManager;
+        this.jwtService = jwtService;
+    }
+
+    @GetMapping("/users")
+    public List<UserDTO> getAllUsers() {
+        return userService.getAllUsers();
+    }
+
+    @GetMapping("/users/{id}")
+    public UserDTO getUser(@PathVariable Long id) {
+        return userService.getUser(id);
+    }
+
+    @PostMapping("/signup")
+    public ResponseEntity<ErrorDTO> registerUser(@RequestBody SignUpDTO signUpDTO) {
+        if (userRepository.existsByUsername(signUpDTO.getUsername())) {
+            ErrorDTO errorDTO = new ErrorDTO(HttpStatus.BAD_REQUEST.value(), "Username already exists!", System.currentTimeMillis());
+            return new ResponseEntity<>(errorDTO, HttpStatus.BAD_REQUEST);
+        }
+        if (userRepository.existsByEmail(signUpDTO.getEmail())) {
+            ErrorDTO errorDTO = new ErrorDTO(HttpStatus.BAD_REQUEST.value(), "Email already exists!", System.currentTimeMillis());
+            return new ResponseEntity<>(errorDTO, HttpStatus.BAD_REQUEST);
+        }
+        userService.createAndSaveUser(signUpDTO);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+    @PostMapping("/login")
+    public JwtResponseDTO AuthenticateAndGetToken(@RequestBody AuthRequestDTO authRequestDTO) {
```

Comment:
If you have the name Request, Response in the DTO, I will try to match those. In this case, I would assume the following classes:

JwtRequestDTO - JwtResponseDTO
AuthRequestDTO - AuthResponseDTO

But in this case it feels for me, like we mismatched the request and response for no reason. A more favorable naming of these DTOs, for example:

AuthRequestDTO - AuthResponseDTO or JwtAuthResponseDTO

Reply:
I actually have LoginDTO that I don't use with same parameters as AuthRequestDTO. So I deleted the AuthRequestDTO as redundant and used LoginDTO. It should be more clean now.